core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackDownscaleValidatorService.java

```diff
@@ -0,0 +1,61 @@
+package com.sequenceiq.cloudbreak.service.stack;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.model.user.IdentityUser;
+import com.sequenceiq.cloudbreak.controller.BadRequestException;
+import com.sequenceiq.cloudbreak.controller.NotFoundException;
+import com.sequenceiq.cloudbreak.domain.InstanceMetaData;
+import com.sequenceiq.cloudbreak.repository.InstanceMetaDataRepository;
+import com.sequenceiq.cloudbreak.service.AuthorizationService;
+
+@Service
+public class StackDownscaleValidatorService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackDownscaleValidatorService.class);
+
+    @Inject
+    private InstanceMetaDataRepository instanceMetaDataRepository;
+
+    @Inject
+    private AuthorizationService authorizationService;
+
+    public void checkUserHasRightToModify(IdentityUser user) {
+        try {
+            authorizationService.hasWritePermission(user);
+        } catch (AccessDeniedException hasNoRight) {
+            String message = String.format("Private stack is only modifiable by the owner. – [%s]", hasNoRight.getMessage());
+            LOGGER.warn(message);
+            throw new BadRequestException(message, hasNoRight);
+        }
+    }
+
+    public void checkInstanceIsTheAmbariServerOrNot(@Nonnull String instancePublicIp, @Nonnull String instancePrivateIp, String ambariIp, Long stackId) {
+        requireNonNull(instancePublicIp);
+        requireNonNull(instancePrivateIp);
+        if (Objects.equals(instancePublicIp, ambariIp) && getGatewayPrivateIp(stackId).equals(instancePrivateIp)) {
```

Comment:
@keyki 
Can we allow users to delete/terminate nodes that runs a standby Ambari server?

Reply:
I don't think we should allow terminating standby Ambari server nodes.